### PR TITLE
[enhancement] LAPW functions

### PR DESCRIPTION
**Pull request by toxa81**

### CODE REVIEW

File: src/context/simulation_context.cpp [hunk 1]
```diff
@@ -1397,59 +1266,59 @@ Simulation_context::init_atoms_to_grid_idx(double R__)
...
-    }
-    if (cfg().control().print_checksum()) {
-        auto z1 = theta_pw_.checksum();
-        auto d1 = theta_.checksum();
-        mpi::Communicator(spfft<double>().communicator()).allreduce(&d1, 1);
-        utils::print_checksum("theta", d1, this->out());
-        utils::print_checksum("theta_pw", z1, this->out());
-    }
-}
+//void
```
Comment (mtaillefumier) [inline src/context/simulation_context.cpp:1269]: we can remove this commented out code.

File: src/function3d/make_periodic_function.hpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+/// Make periodic function out of form factors.
+/** Return vector of plane-wave coefficients */
+template <sddk::index_domain_t index_domain, typename F>
+inline auto
+make_periodic_function(Unit_cell const& uc__, fft::Gvec const& gv__,
+        sddk::mdarray<std::complex<double>, 2> const& phase_factors_t__, F&& form_factors__)
+{
+    PROFILE("sirius::make_periodic_function");
+
+    double fourpi_omega = fourpi / uc__.omega();
```
Comment (mtaillefumier) [inline src/function3d/make_periodic_function.hpp:39]: const double instead of double

File: src/function3d/make_periodic_function.hpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    auto ngv = (index_domain == sddk::index_domain_t::local) ? gv__.count() : gv__.num_gvec();
+    sddk::mdarray<std::complex<double>, 1> f_pw(ngv);
+    f_pw.zero();
+
+    #pragma omp parallel for schedule(static)
+    for (int igloc = 0; igloc < gv__.count(); igloc++) {
+        /* global index of G-vector */
+        int ig   = gv__.offset() + igloc;
```
Comment (mtaillefumier) [inline src/function3d/make_periodic_function.hpp:48]: const int and const double.

**Review by mtaillefumier — APPROVED** [review]

**Comment by toxa81** [comment]
cscs-ci run default

**Pull request merged**